qemu/tests/slof_device_tree.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from autotest.client.shared import utils, error
+
+from virttest import utils_misc
+
+from virttest.qemu_devices import qcontainer
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    slof_device_tree test:
+    steps:
+    1. Boot up a guest.
+    2. Check the guest device tree information.
+    3. Compare the value with expected result.
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_guest_system_id(vm_session):
+        """
+        Get guest system id.
+
+        :param vm_session: session to checked vm.
+        :return: [guest system id]
+        """
+        (status, output) = vm_session.cmd_status_output(
+            "echo `cat /proc/device-tree/system-id`")
+        if status != 0:
+            raise error.TestError("Failed to get guest system id")
+        return output.strip()
+
+    def get_guest_uuid(vm_session):
+        """
+        Get guest uuid.
+
+        :param vm_session: session to checked vm.
+        :return: [guest uuid]
+        """
+        (status, output) = vm_session.cmd_status_output(
+            "echo `cat /proc/device-tree/vm,uuid`")
+        if status != 0:
+            raise error.TestError("Failed to get guest system id")
+        return output.strip()
+
+    def get_host_system_id(vm_session):
+        """
+        Get host system id.
+
+        :param vm_session: session to checked vm.
+        :return: [host system id]
+        """
+        (status, output) = vm_session.cmd_status_output(
+            "echo `cat /proc/device-tree/host-serial`")
+        if status != 0:
+            raise error.TestError("Failed to get host serial from guest")
+        return output.strip()
+
+    def get_guest_partition_name(vm_session):
```

Comment:
There are several func() are same formats,
I think it's better to extract comment part and write a common func() for it.